### ISSUE TITLE
Windows chef-client service should be run as Administrator

### ISSUE THREAD

**Issue by docwhat** [chef/chef#2861]
The windows chef-client currently runs as the Local System user.  This includes whether it is installed via the `.msi` or via the chef-client cookbook.

I feel it should run chef-client as Administrator for a couple reasons:

1. 2008r2 and 2012 have very different permissions for Local System. e.g. in 2008r2, you cannot configure `winrm` unless you're the Administrator; in 2012 you can.
2. If I'm troubleshooting things, a users instinct is to run `chef-client` from a powershell terminal or cmd terminal.  This will behave differently than the service, making debugging problems much more difficult.
3. In Linux, the service is run as root.  Why shouldn't Windows do the equivalent?

Downsides: I know that chef-client cookbook will have to ask for the Administrator password and that the `.msi` installer will have to ask too.  I think this is acceptable.

**Comment by smurawski** [comment]
TL;DR - Running as a scheduled task (supported by the chef-client cookbook) will allow you to run as a different user and in a context that is easier to troubleshoot.

To answer your points -

1 - Yep, 2008 R2 and 2012 have modified security models, and whatever comes next will likely change things as well.  Given that the in-box configuration management agent (Desired State Configuration) runs as local system, and that the resources Microsoft builds to support their configuration issues will leverage that agent, Local System is the right default for Chef Client.
2 - Services operate in a different context.  Even running the service as Administrator will still have some differences in the service context.  Scheduled tasks more readily replicate a logged-in user context and the chef-client cookbook supports setting up Chef to run as a scheduled task - where you can supply the user account and password.
3 - In Windows, Local System is effectively root. 

In all, if the goal is to run in a known "real" user context (for troubleshooting, access to system components, whatever), using a scheduled task rather than the service makes more sense.

**Comment by docwhat** [comment]
Do you have a link to the scheduled task docs?

**Comment by galenemery** [comment]
@docwhat Here's an export of my scheduled task I just created: https://gist.github.com/galenemery/9c1df375f49f021094f8

TL;DR go to scheduled tasks, create a task that runs "chef-client" with admin privileges.  Set it to run as often as you'd like.

**Comment by docwhat** [comment]
@galenemery I just found the [chef-client `task` recipe](https://github.com/opscode-cookbooks/chef-client/blob/master/recipes/task.rb) that seems to do this... combined with some powershell to remove the service (though I need to figure out if I'm *running* as the service) I can switch them all over to scheduled tasks.

**Comment by galenemery** [comment]
@docwhat I was pretty certain we had that.  Perfect!

Now if just that was well documented.

To find out if they're running as a service, just check if you're running the chef-client default or windows_service recipes.

**Comment by btm** [comment]
https://getchef.zendesk.com/hc/en-us/articles/205233360-Should-I-run-chef-client-on-Windows-as-a-service-or-a-scheduled-task-
